lib/cli/daemoncommand.cpp

```diff
@@ -413,6 +413,7 @@ static void WorkerSignalHandler(int num, siginfo_t *info, void*)
 			if (info->si_pid == 0 || info->si_pid == l_UmbrellaPid) {
 				// The umbrella process requested our termination
 				Application::RequestShutdown();
+				(void)signal(num, SIG_DFL);
```

Comment:
Shouldn't this reset both signals at once? Otherwise you could do `kill -INT $pid; kill -TERM $pid` and still run `Application::RequestShutdown()` twice.

Also why do you use `SIG_DFL` instead of `SIG_IGN`? The default action for SIGINT and SIGTERM is to immediately terminate the process. Don't we want to wait for the actual shutdown to finish?

When trying to verify that claim, I noticed that the umbrella process already immediately shuts down when receiving one of these signals twice as it already resets to `SIG_DFL`:
https://github.com/Icinga/icinga2/blob/91ffa89dad41fce6d9feee1209bb8273dd523cf0/lib/cli/daemoncommand.cpp#L367-L381

How did you observe the behavior demonstrated in your comment? Does this behave differently on macOS? (I think it shouldn't the default behavior for these signals is standardized in POSIX).

Reply:
Unfortunately I don’t know _how exactly_ this occurred.

Imagine you suddenly get this behavior, press ^C repeatedly, but it doesn’t terminate – _of course_ we don’t want to wait.